#!/bin/bash
#PBS-l nodes=2
#PBS-l walltime=1000:00:00
#PBS -j oe

set -e
echo "START" $(date)

#8_8_17

module load StringTie/1.3.3b-foss-2016b

# This script takes bam files from HISAT (processed by SAMtools) and performs StringTie assembly and quantification

# StringTie to assemble transcripts for each sample with the GFF3 annotation file
cd /data3/marine_diseases_lab/erin/Bio_project_SRA/pipeline_files/

array1=($(ls $F/*.bam))

for i in ${array1[@]}; do
	stringtie -G /data3/marine_diseases_lab/erin/Bio_project_SRA/pipeline_files/Crassostrea_gigas.gff -B -e -o ${i}.gtf â€“l  $(echo ${i}|sed "s/\..*//")) ${i}
	echo "${i}"
done 
	# -o specifies the output name
	# -G specifies you are aligning with an option GFF or GTF file as well to perform novel 
	# transcript discovery 
	# -l Sets <label> as the prefix for the name of the output transcripts. Default: STRG
	# -B this enables the output to be in the format of Ballgown input files, and StringTie can be used as as direct replacement for the tablemaker program in Ballgown
	
# StringTie Merge, will merge all GFF files and assemble transcripts into a non-redundant set of transcripts, after which re-run StringTie merge

stringtie --merge 

stringtie -eB

# gffcompare 

# Use provided python script (prepDE.py) to extract this read count information directly from the files generated by StringTie (run with the -e parameter).

# Helpful reference website