---
title: "systemPipeR_apoptosis_pipeline"
output: pdf_document
---
## Collect SRA data from NCBI:

Used two different scripts in the URI cluster in order to process paired end read files and single end files.

  For single end reads: fetchSRA.sh
  For paired end reads: fetchSRA_paired.sh
  
  When using, make sure the working directory has *.txt endings for only the SE read files, and *_paired.text for only the PE read files. These file endings are used as filters in the script.


Install bioconductor
```{r}
#install bioconductor
source( "http://www.bioconductor.org/biocLite.R") 
biocLite("BiocUpgrade") 
biocLite(c("ShortRead","DESeq", "edgeR") )
```



## Load systemPipeR library and documentation
```{r}
source("http://bioconductor.org/biocLite.R") # Sources the biocLite.R installation script 
biocLite("systemPipeR") # Installs systemPipeR 
biocLite("systemPipeRdata") # Installs systemPipeRdata
library("systemPipeR") # Loads the package
#library(help="systemPipeR") # Lists package info

library(systemPipeRdata)
genWorkenvir(workflow="rnaseq", mydirname = /data3/marine_diseases_lab/erin/Bio_project_SRA/PE_fastq) #change to cluster directory 
setwd("rnaseq")
```

##Access the targets file

Access the targets file (SE) samples, Construct SYSargs object from param and targets files.
```{r}
#library(systemPipeR)
#targetspath <- system.file("extdata", "targets_SE.txt", package="systemPipeR")
#read.delim(targetspath, comment.char = "#")
#Path specified in the targets_SE.txt file

```


## Access the targets file (PE) samples (second time running through script)
```{r}
targetspath <- system.file("extdata", "targets_PE.txt", package="systemPipeR")
read.delim(targetspath2, comment.char = "#")[1:2,1:6]
```

## Sample Comparisons for PE
Sample comparisons are defined in the header lines of the targets file starting with ‘# <CMP>’.

```{r }
readLines(targetspath)[1:5] #set because the CMPs are listed rows 1-5
```

The function readComp imports the comparison information and stores it in a list. 
```{r }
readComp(file=targetspath, format="vector", delim="-")
```

## Construct Sysargs object
```{r }
args <- systemArgs(sysma="param/trim.param", mytargets="targets_PE.txt")
```

## Read Pre-processing
The paths to the resulting output FASTQ files are stored in the outpaths slot of the SYSargs object. 

1. bbtools_trim_paired.sh run in cluster to trim adapters off the end of reads, to perform quality trimming at the ends of reads with quality less than Phred score 10, and quality filter out all sequences with a quality below phred score 10. Path to these pre-processed files is used in the targets_PE.txt file.

This script also produces histogram reports of each run containing information on: base frequency, quality scores, gc content, average quality, and length.

## Alignment with Tophat2

Tophat2 will be used to map reads to the Crassostrea gigas reference genome available from Ensembl. 
Cufflinks will then be used to perform transcript identification and quantification using the GFF3 file for C. gigas available at Ensembl at the following web address: http://metazoa.ensembl.org/Crassostrea_gigas/Info/Index

Build Bowtie2 index:
```{r}
args <- systemArgs(sysma="param/tophat.param", mytargets="targets_PE.txt")
#moduleload(modules(args)) # Skip if module system is not available
system("bowtie2-build /data3/marine_diseases_lab/erin/Bio_project_SRA/PE_fastq/GCA_000297895.1.fasta /data3/marine_diseases_lab/erin/Bio_project_SRA/PE_fastq/GCA_000297895.1.fasta")

bampaths <- runCommandline(args=args)

```

Run tophat2 ? 
``` {r}

```

Run Cufflinks
```{r}
```


## Read and Alignment Count stats
Table of Read and Alignment Count stats
``` {r}
read_statsDF <- alignStats(args) 
write.table(read_statsDF, "results/alignStats.xls", row.names=FALSE, quote=FALSE, sep="\t")
```

## Read counting for mRNA profiling experiments

Need to create a txdb 
```{r}
library(GenomicFeatures)
txdb <- makeTxDbFromGFF(file="/data3/marine_diseases_lab/erin/Bio_project_SRA/PE_fastq/GCA_000297895.1.gff3", format="gff3", dataSource="Ensembl", organism="C. gigas")
saveDb(txdb, file="/data3/marine_diseases_lab/erin/Bio_project_SRA/PE_fastq/GCA_000297895.1.sqlite")
```

## Differential Gene Expression Analysis using DESeq2
1. Correlation analysis of samples
```{r}
library(DESeq2, warn.conflicts=FALSE, quietly=TRUE); library(ape, warn.conflicts=FALSE)
countDFpath <- system.file("extdata", "countDFeByg.xls", package="systemPipeR")
countDF <- as.matrix(read.table(countDFpath))
colData <- data.frame(row.names=targetsin(args)$SampleName, condition=targetsin(args)$Factor)
dds <- DESeqDataSetFromMatrix(countData = countDF, colData = colData, design = ~ condition)
d <- cor(assay(rlog(dds)), method="spearman")
hc <- hclust(dist(1-d))
plot.phylo(as.phylo(hc), type="p", edge.col=4, edge.width=3, show.node.label=TRUE, no.margin=TRUE)
```

2. DEG analysis with DESeq2
The following run_DESeq2 function is a convenience wrapper for identifying DEGs in batch mode with DESeq2 (Love, Huber, and Anders 2014) for any number of pairwise sample comparisons specified under the cmp argument. 

```{r}
degseqDF <- run_DESeq2(countDF=countDFeByg, targets=targets, cmp=cmp[[1]], independent=FALSE)
```

Code to filter and plot DEG results
```{r}
DEG_list2 <- filterDEGs(degDF=degseqDF, filter=c(Fold=2, FDR=10))
```

## Heatmap analysis
"The following example performs hierarchical clustering on the rlog transformed expression matrix subsetted by the DEGs identified in the above differential expression analysis. It uses a Pearson correlation-based distance measure and complete linkage for cluster joining."

``` {r}
library(pheatmap)
geneids <- unique(as.character(unlist(DEG_list[[1]])))
y <- assay(rlog(dds))[geneids, ]
pdf("heatmap1.pdf")
pheatmap(y, scale="row", clustering_distance_rows="correlation", clustering_distance_cols="correlation")
dev.off()
```
